test/core/xds/xds_transport_grpc_test.cc

```diff
@@ -0,0 +1,210 @@
+  // Set type_url.
+  std::string type_url_str = absl::StrCat("type.googleapis.com/", type_url);
+  req.set_type_url(type_url_str);
+  // Add resource_names.
+  for (const std::string& resource_name : resource_names) {
+    *req.add_resource_names() = resource_name;
+  }
+  return req.SerializeAsString();
+}
+
+using EventHandlerEvent = absl::variant<bool, absl::Status, std::string>;
+
+class AdsService : public envoy::service::discovery::v3::
+                       AggregatedDiscoveryService::Service {
+ public:
+  grpc::Status StreamAggregatedResources(
+      grpc::ServerContext* /* context */,
+      grpc::ServerReaderWriter<envoy::service::discovery::v3::DiscoveryResponse,
+                               envoy::service::discovery::v3::DiscoveryRequest>*
+          stream) override {
+    gpr_log(GPR_DEBUG, "In here!");
+    envoy::service::discovery::v3::DiscoveryRequest req;
+    stream->Read(&req);
+    gpr_log(GPR_DEBUG, "%s", req.DebugString().c_str());
+    envoy::service::discovery::v3::DiscoveryResponse res;
+    stream->WriteLast(res, grpc::WriteOptions());
+    return grpc::Status(grpc::StatusCode::UNIMPLEMENTED, "Not implemented yet");
+  }
+};
+
+class TestEventHandler
+    : public XdsTransportFactory::XdsTransport::StreamingCall::EventHandler {
+ public:
+  explicit TestEventHandler(std::vector<EventHandlerEvent>* events)
+      : events_(events) {}
+
+  void OnRequestSent(bool ok) override {
+    gpr_log(GPR_DEBUG, "on request sent %d", ok);
+    events_->emplace_back(ok);
+  }
+
+  void OnRecvMessage(absl::string_view payload) override {
+    gpr_log(GPR_DEBUG, "message recv");
+    events_->emplace_back(std::string(payload));
+  }
+
+  void OnStatusReceived(absl::Status status) override {
+    gpr_log(GPR_DEBUG, "status %s", std::string(status.message()).c_str());
+    events_->emplace_back(std::move(status));
+  }
+
+ private:
+  std::vector<EventHandlerEvent>* events_;
+};
+
+class AdsServer {
+ public:
+  AdsServer() : server_thread_(ServerThread, this) {
+    MutexLock lock(&mu_);
+    if (url_.empty()) {
+      cond_.Wait(&mu_);
+    }
+  }
+
+  ~AdsServer() {
+    server_->Shutdown();
+    server_thread_.join();
+  }
+
+  std::string url() {
+    MutexLock lock(&mu_);
+    return url_;
+  }
+
+ private:
+  static void ServerThread(AdsServer* ads_server) { ads_server->Run(); }
+
+  void Run() {
+    {
+      MutexLock lock(&mu_);
+      url_ = absl::StrFormat("localhost:%d", grpc_pick_unused_port_or_die());
+      grpc::ServerBuilder builder;
+      builder.AddListeningPort(url_, grpc::InsecureServerCredentials());
+      builder.RegisterService(&ads_service_);
+      server_ = builder.BuildAndStart();
+      cond_.SignalAll();
+    }
+    server_->Wait();
+  }
+
+  std::thread server_thread_;
+  Mutex mu_;
+  CondVar cond_;
+  std::string url_ ABSL_GUARDED_BY(mu_);
+  AdsService ads_service_ ABSL_GUARDED_BY(mu_);
+  std::unique_ptr<grpc::Server> server_;
+};
+
+TEST(GrpcTransportTest, DISABLED_WaitsWithAdsRead) {
```

Comment:
Is this test going to stay here?  I'm assuming it's not ready for review yet.

Reply:
Removed (to branch on my PC)